spec/index.bs

```diff
@@ -192,34 +192,81 @@ dictionary GPULimits {
 </script>
 
 
-Buffers {#buffers}
-==================
+# {{GPUBuffer}} # {#GPUBuffer}
 
-## GPUBuffer ## {#buffer}
+A {{GPUBuffer}} represents a linear allocation of memory to hold data that can be used by GPU operations.
+Each byte of the allocation can be addressed by its offset from the start of the {{GPUBuffer}}.
 
-<script type=idl>
-interface GPUBuffer : GPUObjectBase {
-    Promise<ArrayBuffer> mapReadAsync();
-    Promise<ArrayBuffer> mapWriteAsync();
-    void unmap();
+{{GPUBuffer}} has the following internal slots:
 
-    void destroy();
-};
-</script>
+<dl dfn-type=attribute dfn-for="GPUBuffer">
+	: <dfn>\[[size]]</dfn>
+	::
+        The Length of the {{GPUBuffer}} allocation in bytes, which is an unsigned long long.
+
+	: <dfn>\[[usage]]</dfn>
+	::
+        The allowed usages for this {{GPUBuffer}}, represented as a bitset unsigned long.
+
+    : <dfn>\[[state]]</dfn>
+    ::
+        This {{GPUBuffer}}'s state which is one of `"unmapped"`, `"mapped"`, `"destroyed"`.
+</dl>
+
+## Creation ## {#buffer-creation}
 
-### Creation ### {#buffer-creation}
+A {{GPUBuffer}} can be created unmapped using the {{GPUDevice/createBuffer(descriptor)}} method.
+
+### {{GPUBufferDescriptor}} ### {#GPUBufferDescriptor}
+
+This specifies the options to use in creating a {{GPUBuffer}}.
+The {{GPUBufferDescriptor/size}} and {{GPUBufferDescriptor/usage}} members are required.
+A {{NotFoundError}} exception MUST be thrown if any of the required members are not specified.
```

Comment:
I think the right error here is TypeError, not NotFoundError.
That said, the WebIDL spec already specifies the behavior in step 4.1.5 here:
https://heycam.github.io/webidl/#es-dictionary
and I don't think we need to (or should, redundantly) specify it, since it's in the IDL definition.